crates/wash-lib/src/context/fs.rs

```diff
@@ -74,67 +98,93 @@ impl ContextDir {
             .into_iter()
             .find(|p| p.file_stem().unwrap_or_default() == name))
     }
+}
+
+fn default_context_dir() -> Result<PathBuf> {
+    Ok(cfg_dir()?.join(DEFAULT_CTX_DIR_NAME))
+}
 
-    fn index_path(&self) -> PathBuf {
-        self.0.join(INDEX_JSON)
+fn initialize_context_dir(context_dir: &Path, default_path: &PathBuf) -> Result<()> {
+    let mut default_context_name = HOST_CONFIG_NAME.to_string();
+
+    // TEMPORARY (TM): look for and parse existing index.json, to preserve backwards compatibility
+    if let Ok(index_file) = File::open(context_dir.join("index.json")) {
+        #[derive(serde::Deserialize)]
+        struct DefaultContext {
+            name: String,
+        }
+
+        if let Ok(old_default_context) =
+            serde_json::from_reader::<_, DefaultContext>(BufReader::new(index_file))
+        {
+            default_context_name = old_default_context.name;
+        }
     }
+    // END TEMPORARY (TM)
+
+    let mut default_file =
+        File::create(default_path).context("failed to create default context file")?;
+    default_file
+        .write_all(default_context_name.as_bytes())
+        .context("failed to write to default context file")?;
```

Comment:
Do you like the explicitness of the errors here w/ failure to create files  and/or write to them vs `std::fs::write`?

Also, it would be great to have more details about the paths that couldn't be created/written

Reply:
To be honest, no, the only reason most of this code didn't change more is because I didn't originally intend to change any of this code 😅 Fixed